relayer/chain/ethereum/header_cache_state.go

```diff
@@ -49,30 +138,39 @@ type HeaderCacheState struct {
 func NewHeaderCacheState(
 	eg *errgroup.Group,
 	initBlockHeight uint64,
-	cl EthashproofCacheLoader,
+	bl BlockLoader,
+	ecl EthashproofCacheLoader,
 ) (*HeaderCacheState, error) {
-	cacheState := EthashproofCacheState{
+	blockCache := NewBlockCache(5)
```

Comment:
Is `5` a good choice here? I feel it's a bit low for a real chain with latency and forks.

Reply:
It might very well need to be adjusted. Fwiw, this cache is not used by the header forwarding / syncing logic. I might use it for that, but with significant changes. My understanding is that, if we want to track forks, we will need to keep the latest 35 heights in a cache. Does that make sense to you @philipstanislaus?